openapiv3/OpenAPIv3.go

```diff
@@ -4373,12 +4337,22 @@ func NewSecurityRequirement(in interface{}, context *compiler.Context) (*Securit
 		message := fmt.Sprintf("has unexpected value: %+v (%T)", in, in)
 		errors = append(errors, compiler.NewError(context, message))
 	} else {
-		allowedKeys := []string{}
-		allowedPatterns := []*regexp.Regexp{pattern4}
-		invalidKeys := compiler.InvalidKeysInMap(m, allowedKeys, allowedPatterns)
-		if len(invalidKeys) > 0 {
-			message := fmt.Sprintf("has invalid %s: %+v", compiler.PluralProperties(len(invalidKeys)), strings.Join(invalidKeys, ", "))
-			errors = append(errors, compiler.NewError(context, message))
+		// repeated NamedStringArray additional_properties = 1;
+		// MAP: StringArray
+		x.AdditionalProperties = make([]*NamedStringArray, 0)
+		for _, item := range m {
+			k, ok := compiler.StringValue(item.Key)
+			if ok {
```

Comment:
Suggestion: put a `continue` in a `!ok` conditional rather than have the "green path" code be indented.

```go
if !ok {
  continue
}

v := item.Value
// ...
```

Reply:
as above, since this is generated code, I'd like to address it separately in issue #167